bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/DeviceThingHandler.java

```diff
@@ -120,13 +120,15 @@ public void initialize() {
 
         for (Channel channel : getThing().getChannels()) {
             KNXChannel knxChannel = KNXChannelFactory.createKnxChannel(channel);
-            knxChannels.put(channel.getUID(), knxChannel);
-            groupAddresses.addAll(knxChannel.getAllGroupAddresses());
 
             if (knxChannel.getChannelType().startsWith("number")) {
                 // check if we need to update the accepted item-type
                 List<InboundSpec> inboundSpecs = knxChannel.getAllGroupAddresses().stream()
                         .map(knxChannel::getListenSpec).filter(Objects::nonNull).map(Objects::requireNonNull).toList();
+                if (inboundSpecs.isEmpty()) {
+                    logger.warn("Skipping {}: configuring a DPT is mandatory on 'number' channels!", channel.getUID());
```

Comment:
Isn't the message a bit misleading? DPTs are not required for a number - in this case the ```KNXChannel``` can be added as it is.
Root cause for the IOOB exception is the failed parsing of the inboundspec, so there is no ```InboundSpec```. Just tried to include a number channel with ```ga=""```` in my config and got the IOOB in 4.0.3 (without this fix).

Reply:
So the first point is: in case inboundSpecs is empty -> there is **no GA** configured, no action, continue with next channel. IOOB fixed.

Second thing: ```inboundSpecs.get(0).getDPT()``` will give you 9.001 for all number items without a configured DPT. The implementation in KNXChannel & TypeNumber seems to ensure we have the default 9.001. I don't see that we need a special handling for this case. Do I overlook something?

Reply:
Maybe we could omit whole new block if we use

```
    String dpt = inboundSpecs.isEmpty() ? knxChannel.getDefaultDPT() : inboundSpecs.get(0).getDPT();
```

WDYT?

Reply:
Does it make sens to have a KNXChannel to be created if the channel config does not contain a single GA?

Reply:
My understanding is that "no configured DPT" means "use default DPT", so it is fine not to configure the DPT. For correctly configuring the channel-type the DPT is sufficient.

In general I can't think of a use-case where no GA is configured and it should probably lead to an error (and skipping the channel when creating the `KNXChannel`). But that seems to be a different issue.